hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java

```diff
@@ -3876,10 +3873,9 @@ public boolean hasPathCapability(final Path path, final String capability)
       throws IOException {
     // qualify the path to make sure that it refers to the current FS.
     final Path p = makeQualified(path);
-    Optional<Boolean> cap = DfsPathCapabilities.hasPathCapability(p,
-        capability);
-    if (cap.isPresent()) {
-      return cap.get();
+    if (DfsPathCapabilities.hasPathCapability(p, capability)
+        && supportsSymlinks()) {
```

Comment:
why we are adding `supportsSymlinks()`? capablity can be anything different from `FS_SYMLINKS`

Reply:
Had added `supportsSymlinks()` returns false to just assert the behaviour at `super.hasPathCapablity()`